### PR TITLE
paginate chat

### PR DESCRIPTION
**Pull request by ryanatkn**
Adds a new paginated query system and implements it just for the `Chat`.

Future:

- fix the problem detailed in the todos
- implement for other views
- merge behavior with `QueryEntities`
- autoload on scroll? (need to handle the case where you can't scroll because not enough loaded to cause a scrollbar)

### CODE REVIEW

File: src/lib/util/api.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import type {Result} from '@feltcoop/util';
```
Comment (ryanatkn) [inline src/lib/util/api.ts:1]: not sure about this module name/location but it seems better than `$lib/server/api`?

File: src/lib/vocab/entity/entityMutations.ts [hunk 2]
```diff
@@ -61,11 +61,27 @@ export const QueryEntities: Mutations['QueryEntities'] = ({ui: {queryByKey}, dis
...
+				} else {
+					query.status.set('failure');
+					query.error.set(result.message);
+				}
+			},
+			() => {
+				if (!query) return;
+				query.status.set('failure');
+				query.error.set('unknown error');
+			},
```
Comment (ryanatkn) [inline src/lib/vocab/entity/entityMutations.ts:84]: this is unrelated to the PR, it fixes error handling of the existing query system that doesn't try to handle pagination

**Review by greatbacon — APPROVED** [review]
